src/reducers/transactionReducer.ts

```diff
@@ -15,14 +15,29 @@
 // along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
 
 import { TransactionActionTypes } from '../actions/transactionActions';
-import type { TransactionsActionType, TransactionState } from '../types/transactionTypes';
+import type { TransactionsActionType, TransactionState, UpdatedTransanctionStatus } from '../types/transactionTypes';
+
+const updateTransactionObject = (state: TransactionState, updatedValues: UpdatedTransanctionStatus) => {
+  const newState = { ...state };
+  if (state.currentTransaction) {
+    Object.keys(updatedValues).map((key) => {
+      newState.currentTransaction![key] = updatedValues[key];
+    });
+  }
```

Comment:
This can be a single statement and IMHO it's mor readable than the loop. The fact that we receive `updatedValues` in case there is no `currentTransaction` is imo a bug, and we should prevent it in some other place.

```suggestion
  return {
     ...state,
     currentTransaction: {
        ...state.currentTransaction,
        updatedValues,
     }
  };
```